tests/unit/s2n_tls13_handshake_state_machine_test.c

```diff
@@ -516,24 +624,39 @@ int main(int argc, char **argv)
         /* Ensure OCSP_STATUS is set by setting the connection status_type */
         conn->status_type = S2N_STATUS_REQUEST_OCSP;
 
-        /* Verify setup: tls1.2 DOES set the flags */
+        /* Verify that tls1.2 DOES set the flags allowed by tls1.2 */
         conn->actual_protocol_version = S2N_TLS12;
         EXPECT_SUCCESS(s2n_conn_set_handshake_type(conn));
         EXPECT_TRUE(conn->handshake.handshake_type & TLS12_PERFECT_FORWARD_SECRECY );
         EXPECT_TRUE(conn->handshake.handshake_type & OCSP_STATUS );
         EXPECT_TRUE(conn->handshake.handshake_type & WITH_SESSION_TICKET );
         EXPECT_TRUE(conn->handshake.handshake_type & CLIENT_AUTH );
 
-        /* Verify that tls1.3 ONLY sets the CLIENT_AUTH flag */
+        /* Verify that tls1.2 DOES NOT set the flags only allowed by tls1.3 */
+        EXPECT_FALSE(conn->handshake.handshake_type & MIDDLEBOX_COMPAT);
+
```

Comment:
HELLO_RETRY_REQUEST is also not allowed with tls1.2, so either move that test to here or update comment to verify that tls1.2 does not set the flag for MIDDLEBOX_COMPAT.

Reply:
Sure, I can check for HELLO_RETRY_REQUEST here. However, there's a whole separate test for HELLO_RETRY_REQUEST because HELLO_RETRY_REQUEST is actually set outside of this method. Not even TLS1.3 sets HELLO_RETRY_REQUEST here. 

The more useful test for HELLO_RETRY_REQUEST is how we handle whether or not the HELLO_RETRY_REQUEST flag is already set when this method is called. The next test covers that.